test/components/routers/test_conditional_router.py

```diff
@@ -305,41 +310,30 @@ def test_custom_filter(self):
             },
         ]
 
-        def get_area_code(phone_num):
-            return int(phone_num.split("-")[0])
-
-        router = ConditionalRouter(routes, custom_filters={"get_area_code": get_area_code})
+        router = ConditionalRouter(routes, custom_filters={"get_area_code": custom_filter_to_sede})
         kwargs = {"phone_num": "123-456-7890"}
         result = router.run(**kwargs)
         assert result == {"good_phone_num": "Phone number has a 123 area code"}
         kwargs = {"phone_num": "321-456-7890"}
         result = router.run(**kwargs)
         assert result == {"bad_phone_num": "Phone number does not have 123 area code"}
 
-    def test_custom_filter_serialization(self):
-        routes = [{"condition": "{{ test|return_true }}", "output": "true", "output_name": "true", "output_type": str}]
-
-        def return_true(*args):
-            return True
-
-        custom_filters = {"return_true": return_true}
+    def test_sede_with_custom_filter(self):
+        routes = [
+            {
+                "condition": "{{ test|custom_filter_to_sede == 123 }}",
+                "output": "true",
+                "output_name": "test",
+                "output_type": str,
+            }
+        ]
+        custom_filters = {"custom_filter_to_sede": custom_filter_to_sede}
         router = ConditionalRouter(routes, custom_filters=custom_filters)
-        kwargs = {"test": "test"}
+        kwargs = {"test": "123-456-789"}
         result = router.run(**kwargs)
-        assert result == {"true": "true"}
+        assert result == {"test": "true"}
         serialized_router = router.to_dict()
```

Comment:
Let's also check the dictionary here.

Reply:
The test is valid as the `test` route condition returns the string `true` when the filter function returns `123` from the input `123-456-789`, but I updated the route and the return value to be more explicit.